src/app/components/article/article.component.ts

```diff
@@ -0,0 +1,26 @@
+import { Component, Input } from '@angular/core';
+import { ArticlesService } from "../../services/articles.service";
+
+@Component({
+  selector: 'app-article',
+  templateUrl: './article.component.html',
+  styleUrls: ['./article.component.scss']
+})
+export class ArticleComponent {
+  @Input() article;
+
+  isReadMore = true;
+
+  constructor(private articlesService: ArticlesService) {}
+
+  toggleArticleContentLength() {
+    this.isReadMore = !this.isReadMore;
+  }
+
+  increaseLikesAmount() {
+    this.articlesService.updateArticle({
+      ...this.article,
+      likesAmount: this.article.likesAmount + 1
+    })
+  }
+}
```

Comment:
for me, it is better to generate output and inside smart list component inject service and increase likes. But it also could be as a solution